advanced/multichain/rpc-reference/stacks-rpc.mdx

```diff
@@ -6,11 +6,51 @@ description: Stacks JSON-RPC Methods
 
 These are the methods that wallets should implement to handle Stacks transfers and messages via WalletConnect.
 
-## stx_transferStx
+## Core Methods (common)
 
-Simple STX transfer, request a transfer of STX tokens. Wallet signs and executes the transaction.
+### stx_getAddresses
 
-### Request
+Retrieve active account addresses; primarily Stacks-focused.
+
+#### Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "stx_getAddresses",
+  "params": {}
+}
+```
+
+#### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "addresses": [
+      {
+        "symbol": "STX",
+        "address": "SP…"
+      }
+    ]
+  }
+}
+```
+
+**Notes:**
+- Use this first to select the wallet's active address.
+- Filter on `symbol: "STX"`.
```

Comment:
This is usually done by filtering on the first characters of the address which are representations of the different account types. SP for mainnet ST for testnet